### PR TITLE
Documentation overhaul

### PR DESCRIPTION
**Pull request by jacobbaungard**
Overhauls the documentation with a new set of documentation on Github Pages! See individual commits for details.

Note that:
- This depends on https://github.com/ITRS-Group/merlin-container-poller/pull/1 being merged as we are pulling some docs from there directly
- We are not removing old docs such as SPECS, HOWTO, TECHNICAL in this PR. In a later stage we'll need to review these documents and ensure we're not loosing any useful information before removing them
- The documentation is deployed on https://itrs-group.github.io/monitor-merlin/ for review

### CODE REVIEW

File: docs/clustering.rst [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+       |
+       | HOSTGROUP: pollergroup  +----------+
+       = ------------------------| poller01 |
+                                 +----------+
+
+Preparation
+^^^^^^^^^^^^
+
+To begin with, prepare three machines (master01, master02, poller01), with
+Naemon and Merlin installed as per the installation instructions (LINK). Make
```
Comment (akselsjogren) [inline docs/clustering.rst:76]: missing link

File: docs/clustering.rst [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+sure that there are no firewalls blocking port 22 (for SSH) & 15551 (Merlins
+default TCP port). Ensure that you had the root password for all machines at
+hand.
+
+Adding a peer
+^^^^^^^^^^^^^^
+
+We'll start by peering master01 and master02. To begin we need to ensure that
+passwordless SSH connection is possible between the nodes. Merlin includes
+a convince script to setup and install SSH keys across the nodes, that we'll
```
Comment (akselsjogren) [inline docs/clustering.rst:86]: "convenience"

File: docs/clustering.rst [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+                                 +----------+
+
+Preparation
+^^^^^^^^^^^^
+
+To begin with, prepare three machines (master01, master02, poller01), with
+Naemon and Merlin installed as per the installation instructions (LINK). Make
+sure that there are no firewalls blocking port 22 (for SSH) & 15551 (Merlins
+default TCP port). Ensure that you had the root password for all machines at
+hand.
```
Comment (akselsjogren) [inline docs/clustering.rst:79]: I guess also that root login, using password, is required between peers to continue setup as documented. That is not always the case on systems (for good reasons too).

File: docs/clustering.rst [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+``pollergroup``.
+
+SSH connection must be established to both masters, so we start by adding SSH
+keys.
+
+.. code-block:: none
+
+   [root@poller01 ~]# mon sshkey push IP-OF-MASTER01
+   [root@poller01 ~]# mon sshkey push IP-OF-MASTER02
+   [root@master01 ~]# mon sshkey push IP-OF-POLLER01
```
Comment (akselsjogren) [inline docs/clustering.rst:159]: These lines look so similar (I had a hard time distinguishing the hostnames in the prompt even), so it's hard to read them and see the differences. Perhaps they could be grouped into per-node commands or something?

The same applies to other similar blocks with commands.

File: docs/encryption.rst [hunk 5]
```diff
@@ -0,0 +1,77 @@
+Encrypted communication
+========================
+
+Merlin supports encrypting the packets send over the Merlin TCP protocol.
```
Comment (akselsjogren) [inline docs/encryption.rst:4]: `s/send/(transmitted|sent)/`

File: docs/encryption.rst [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+Configuring encryption
+-----------------------
+
+In this example we'll set up encryption between two peers, *peer01*, and *peer02*.
+
+Generating encryption keys
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To setup encryption, we first need to generating a encryption key pair consisting
```
Comment (akselsjogren) [inline docs/encryption.rst:20]: s/generating/generate/

File: docs/encryption.rst [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+of a private key and a public key.
+
+To do so we can make sure of the ``mon merlinkey`` command.
```
Comment (akselsjogren) [inline docs/encryption.rst:23]: make use?

File: docs/encryption.rst [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+You can omit ``--write`` flag if you do not write the change to ``merlin.conf``.
+
+We now need to copy the public keys, to the other node. Note the private key
+should never be shared! We can use rsync over ssh to copy the relevant public
+key.
+
+.. code-block:: none
+
+   [root@peer01 ~]# rsync -a /etc/merlin/key.pub peer02:/etc/merlin/peer1.pub
```
Comment (akselsjogren) [inline docs/encryption.rst:43]: For a single file, perhaps `scp /etc/merlin/key.pub peer02:/etc/merlin/peer1.pub` makes for a simpler example..?

File: docs/encryption.rst [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+   [root@peer02 ~]# rsync -a /etc/merlin/key.pub peer01:/etc/merlin/peer2.pub
+
+
+Adjusting configuration
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+To enable encryption, we must set correct the node config in ``merlin.conf``,
```
Comment (akselsjogren) [inline docs/encryption.rst:50]: "must set correct the" ?

File: docs/encryption.rst [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+You can omit ``--write`` flag if you do not write the change to ``merlin.conf``.
+
+We now need to copy the public keys, to the other node. Note the private key
+should never be shared! We can use rsync over ssh to copy the relevant public
+key.
+
+.. code-block:: none
+
+   [root@peer01 ~]# rsync -a /etc/merlin/key.pub peer02:/etc/merlin/peer1.pub
+   [root@peer02 ~]# rsync -a /etc/merlin/key.pub peer01:/etc/merlin/peer2.pub
```
Comment (akselsjogren) [inline docs/encryption.rst:44]: Target files should be `peer0[12].pub` to match merlin.conf below.

File: docs/clustering.rst (hunk 3 above)
Comment (jacobbaungard) [reply docs/clustering.rst:79]: as in: clarify that SSH root login with password should be allowed?

File: docs/clustering.rst (hunk 4 above)
Comment (jacobbaungard) [reply docs/clustering.rst:161]: I added a blank line so that commands from different systems are grouped together. Makes it a bit better at least.

**Review by akselsjogren — APPROVED** [review]
LGTM

**Pull request merged**